sys/crypto/modes/ccm.c

```diff
@@ -37,7 +37,8 @@ static inline int min(int a, int b)
 int ccm_compute_cbc_mac(cipher_t *cipher, const uint8_t iv[16],
                         const uint8_t *input, size_t length, uint8_t *mac)
 {
-    uint8_t offset, block_size, mac_enc[16] = { 0 };
+    uint8_t block_size, mac_enc[16] = { 0 };
+    uint32_t offset;
```

Comment:
Doesn't this limit the message length to `2^32` and isn't the message limit `2^8L` so `2^64`? I think its OK not to expect messages longer than `2^32` (which is already huge!) buy maybe this should be checked and asserted?

I think the doc should be updated with your changes in #12362 as well but this could be left for a follow up PR.

Reply:
Should I change length from `size_t` to `uint32_t`? That would make the size limitation obvious and lead to a better defined situation.

Reply:
Hmm I think this is something with public API, but I'll handle it in a separate PR. Lets get this fix in, lets wait for #12362 to get in and then we will go with this one as well.